common/src/main/java/org/keycloak/common/util/MultiSiteUtils.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.common.util;
+
+import org.keycloak.common.Profile;
+
+public class MultiSiteUtils {
+
+    public static boolean isMultiSiteEnabled() {
+        return Profile.isFeatureEnabled(Profile.Feature.MULTI_SITE);
+    }
+
+    /**
+     * @return true when user sessions are stored in the database. In multi-site setup this is false when REMOTE_CACHE feature is enabled
+     */
+    public static boolean isPersistentSessionsEnabled() {
+        return Profile.isFeatureEnabled(Profile.Feature.PERSISTENT_USER_SESSIONS) || (isMultiSiteEnabled() && !Profile.isFeatureEnabled(Profile.Feature.REMOTE_CACHE));
```

Comment:
Aren't we always assuming that MULTI_SITE has REMOTE_CACHE? In which case this can just be:

```java
return Profile.isFeatureEnabled(Profile.Feature.PERSISTENT_USER_SESSIONS) || isMultiSiteEnabled();
```

Reply:
We need to have the possibility to configure pure volatile sessions using `REMOTE_CACHE`. With the suggested condition we would always use persistent sessions when `MULTI_SITE` is enabled with no possibility to have `MULTI_SITE + REMOTE_CACHE`.

Also, `MULTI_SITE` on its own has only partial `REMOTE_CACHE`. User and client sessions are not used. 

I see this isn't very clear, however, I don't see any way to simplify this.

Reply:
> we would always use persistent sessions when `MULTI_SITE` is enabled with no possibility to have `MULTI_SITE + REMOTE_CACHE`.

Is that something we want to support initially? I thought the assumption was that for 26.0 at least, we would have MULTI_SITE + PERSISTENT_USER_SESSIONS + REMOTE_CACHE.

Reply:
Depends on what you mean with `REMOTE_CACHE`. 

Our current feature mechanism doesn't allow for composite features therefore, when you enable only `MULTI_SITE`, the rest of the features: `PERSISTENT_USER_SESSIONS` and  `REMOTE_CACHE` are both disabled. 

With Keycloak 26 we are interested in two setups:
1. Multi-site with persistent-sessions and remote-cache for all entities apart from sessions - this is configured (in this PR) with `MULTI_SITE` feature, other features are disabled
2. Multi-site with volatile sessions - here we need to not use persistent sessions but sessions from remote-cache feature - this is currently configured (in this PR) with `MULTI_SITE + REMOTE_CACHE` features. 

The condition in question contains `!Profile.isFeatureEnabled(Profile.Feature.REMOTE_CACHE)` to disable persistent sessions so we can run in mode `2.`. If we go with the condition suggested by you in the first comment, we would not be able to run in mode `2.` because persistent sessions would always be enabled with `MULTI_SITE` profile.

Reply:
Ok it makes sense to me now. Can you add a bit more detail to the method javadoc to make it clear that the MULTI_SITE+REMOTE_CACHE combination is for muliti-site with volatile sessions.